metadata-ingestion/src/datahub/ingestion/source/schema_inference/json.py

```diff
@@ -29,12 +30,18 @@
 
 
 class JsonInferrer(SchemaInferenceBase):
-    def infer_schema(self, file: IO[bytes]) -> List[SchemaField]:
+    def __init__(self, is_newline_json: bool) -> None:
+        super().__init__()
+        self.is_newline_json = is_newline_json
 
-        datastore = ujson.load(file)
+    def infer_schema(self, file: IO[bytes]) -> List[SchemaField]:
+        if self.is_newline_json:
```

Comment:
Is there a way we could automatically make this determination? A few ideas:

- using the extensions `.ndjson` or `.jsonl`
- checking the first character e.g. `{` -> json lines but `[` -> regular json? This doesn't seem foolproof though
- attempt a normal json read, and if it errors with the `"Extra data"` error then try the newline format

Reply:
Updated.

- Replace NewlineJSON with jsonlines.
- Retry with jsonlines Reader when `ujson.load` returns JSONDecodeError.

Reply:
Thanks!